proposals/lip-0053.md

```diff
@@ -246,12 +251,26 @@ def verifyPartnerChainOutboxRoot(ccu: Transaction) -> None:
         inboxTree.append(sha256(ccmBytes))
 
     # calculateRootFromRightWitness is specified in LIP 0031.
-    # Notice that if len(ccu.params.inboxUpdate.messageWitnessHashes) == 0, this function returns the same value as inbox.root,
+    # Notice that if len(ccu.params.inboxUpdate.messageWitnessHashes) == 0, this function returns the same value as inboxTree.root,
     # because all messages have been included in the CCU.
     newInboxRoot = calculateRootFromRightWitness(inboxTree.size, inboxTree.appendPath, ccu.params.inboxUpdate.messageWitnessHashes)
 
-    # Verification for non-empty certificates.
+    
     outboxRootWitness = ccu.params.inboxUpdate.outboxRootWitness
+    # The outbox root witness properties must be set either both to their default values
+    # or both to a non-default value.
+    if outboxRootWitness.bitmap == EMPTY_BYTES and len(outboxRootWitness.siblingHashes) > 0:
+        raise Exception("The bitmap in the outbox root witness must be non-mepty if the sibling hashes are non-empty.")
+    if outboxRootWitness.bitmap != EMPTY_BYTES and len(outboxRootWitness.siblingHashes) == 0:
+        raise Exception("The sibling hashes in the outbox root witness must be non-mepty if the bitmap is non-empty.")
+
+    # The outbox root witness is empty if and only if the certificate is empty
+    if outboxRootWitness.bitmap == EMPTY_BYTES and len(ccu.params.certificate) > 0:
```

Comment:
Doesn't the `verify` function in line 289 already fail in this case? I don't know the exact behavior of this function, but I would expect that.

Reply:
yes, this is correct. However, as @ishantiw pointed out, it is not necessary to do the SMT verification if this condition is not fulfilled (this is basically just a fail shortcut)